### PR TITLE
Added more volume ops

### PR DESCRIPTION
**Pull request by aujjwal-redhat**
Added log_volume_info_nad_status and is_volume_exported.

Fixes: #675

Signed-off-by: Ayush Ujjwal <aujjwal@redhat.com>

### CODE REVIEW

File: common/ops/gluster_ops/volume_ops.py [hunk 1]
```diff
@@ -485,7 +490,10 @@ def get_volume_info(self, node: str = None, volname: str = 'all') -> dict:
 
         cmd = f"gluster volume info {volname} --xml"
 
-        ret = self.execute_abstract_op_node(cmd, node)
+        ret = self.execute_abstract_op_node(cmd, node, excep)
+
+        if not excep and int(ret['msg']['opRet']) != 0:
```
Comment (nik-redhat) [inline common/ops/gluster_ops/volume_ops.py:495]: opRet will have a string value, make it '0'

Comment (nik-redhat) [reply common/ops/gluster_ops/volume_ops.py:495]: Actually, we don't need this an `excep` flag for this function. Just an empty dict check in the caller function shall do, if it doesn't work then something wrong with the command.

File: common/ops/gluster_ops/volume_ops.py [hunk 2]
```diff
@@ -1082,3 +1090,70 @@ def bulk_volume_creation(self, node: str, number_of_volumes: int,
...
+            self.execute_abstract_op_node(cmd, node)
+
+            cmd = ("smbclient -L localhost -U | "
+                   f"grep -i -Fw gluster {volname}")
+            ret = self.execute_abstract_op_node(cmd,
+                                                node,
+                                                False)
+            if ret['error_code'] != 0:
+                return False
+            else:
```
Comment (nik-redhat) [inline common/ops/gluster_ops/volume_ops.py:1157]: Same as above

File: common/ops/gluster_ops/volume_ops.py [hunk 3]
```diff
@@ -1082,3 +1090,70 @@ def bulk_volume_creation(self, node: str, number_of_volumes: int,
...
+            self.execute_abstract_op_node(cmd, node)
+
+            cmd = f"showmount -e localhost | grep -w {volname}"
+            ret = self.execute_abstract_op_node(cmd,
+                                                node,
+                                                False)
+            if ret['error_code'] != 0:
+                return False
+            else:
+                return True
```
Comment (nik-redhat) [inline common/ops/gluster_ops/volume_ops.py:1144]: Remove this part and simply keep a `return True` at the end.

**Review by nik-redhat — CHANGES_REQUESTED** [review]

File: common/ops/gluster_ops/volume_ops.py (hunk 1 above)
Comment (aujjwal-redhat) [reply common/ops/gluster_ops/volume_ops.py:495]: Ack

File: common/ops/gluster_ops/volume_ops.py (hunk 2 above)
Comment (aujjwal-redhat) [reply common/ops/gluster_ops/volume_ops.py:1157]: Ack

File: common/ops/gluster_ops/volume_ops.py (hunk 3 above)
Comment (aujjwal-redhat) [reply common/ops/gluster_ops/volume_ops.py:1144]: Ack

File: common/ops/gluster_ops/volume_ops.py (hunk 1 above)
Comment (aujjwal-redhat) [reply common/ops/gluster_ops/volume_ops.py:495]: But what if the command itself failed to execute. In that case there will be no dict.

Comment (nik-redhat) [reply common/ops/gluster_ops/volume_ops.py:495]: Ah....true. Then, let's keep this check, just update it to compare string.

Comment (aujjwal-redhat) [reply common/ops/gluster_ops/volume_ops.py]: yeah

**Comment by nik-redhat** [comment]
Can you update the code for the `get_vol_status` as well, to not do any logging and return in case of `excep=False` in the same PR?

**Comment by nik-redhat** [comment]
And, with that a couple oof TC needs fixes like `test_gluster_detect_drop_of_outbound_traffic` and `test_volume_status_with_absent_bricks` do fix them as well.

**Comment by aujjwal-redhat** [comment]
> Can you update the code for the `get_vol_status` as well, to not do any logging and return in case of `excep=False` in the same PR?

Ok

**Comment by aujjwal-redhat** [comment]
```js
Summary:
+---------------+-------+--------------+
|    Category   | Cases | Pass Percent |
+---------------+-------+--------------+
| nonDisruptive |   0   |      0       |
|   Disruptive  |   1   |    100.0     |
|     Total     |   1   |    100.0     |
+---------------+-------+--------------+
Table:
 test_volume_status_with_absent_bricks
 
+-------------+-------------+--------------------+
| Volume Type | Test Result |  Time taken (sec)  |
+-------------+-------------+--------------------+
|     dist    |     PASS    | 24.71252703666687  |
|     rep     |     PASS    | 32.111796855926514 |
|   dist-rep  |     PASS    | 34.945164918899536 |
|     disp    |     PASS    | 31.519803524017334 |
|  dist-disp  |     PASS    | 36.63633751869202  |
|     arb     |     PASS    | 32.81458806991577  |
|   dist-arb  |     PASS    | 36.21625828742981  |
+-------------+-------------+--------------------+
```

File: tests/functional/glusterd/test_gluster_detect_drop_of_outbound_traffic.py [hunk 4]
```diff
@@ -81,7 +81,7 @@ def run_test(self, redant):
         # Check volume status command
         vol_status = redant.get_volume_status(self.vol_name, self.server1,
                                               excep=False)
-        if vol_status is None:
+        if ret['msg']['opRet'] != '0':
             raise Exception("Failed to get the vol status")
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_gluster_detect_drop_of_outbound_traffic.py:85]: As per the above comment, it should fail, isn't it?

Comment (aujjwal-redhat) [reply tests/functional/glusterd/test_gluster_detect_drop_of_outbound_traffic.py:85]: yeah True. Ig the logic needs to be changed here. on it

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**